### PR TITLE
i2c: Adds i2c.Addr type and flag.Value interface.

### PR DESCRIPTION
**Pull request by NeuralSpaz**
This is adds a new type i2c.Addr and Set methods that implement the flag.Value interface as a convenience for device maintainers.

Add i2c.Addr
Add i2c.Addr Set()
Add Tests.

helps with #360

### CODE REVIEW

File: conn/i2c/i2c.go [hunk 1]
```diff
@@ -109,6 +110,20 @@ func (d *Dev) Duplex() conn.Duplex {
 	return conn.Half
 }
 
-//
+// Addr implements flag.Value.
```
Comment (maruel) [inline conn/i2c/i2c.go]: It should state first what it represents. :)

File: conn/i2c/i2c.go [hunk 2]
```diff
@@ -109,6 +110,20 @@ func (d *Dev) Duplex() conn.Duplex {
...
+type Addr uint16
+
+// Set sets the Addr to a value respresented by the string s. Values maybe in
```
Comment (maruel) [inline conn/i2c/i2c.go]: It is Set that implements flag.Value.

represented*

File: conn/i2c/i2c.go [hunk 3]
```diff
@@ -109,6 +110,20 @@ func (d *Dev) Duplex() conn.Duplex {
...
+// decimal or hexadecimal form.
+func (a *Addr) Set(s string) error {
+	u, err := strconv.ParseUint(s, 0, 16)
+	if err != nil {
+		return errI2CSetError
+	}
+	*a = Addr(u)
```
Comment (maruel) [inline conn/i2c/i2c.go]: Valid addresses are on 11 bits (when including R/W bit); so values above 0x7ff are invalid.
https://www.totalphase.com/support/articles/200349176-7-bit-8-bit-and-10-bit-I2C-Slave-Addressing gives a nice explanation.

**Comment by NeuralSpaz** [comment]
no problem, Iv'e never used or even seen anything that uses 10 bit addressing, its one of those elusive parts of the spec, that most people will never use.

File: conn/i2c/i2c.go [hunk 4]
```diff
@@ -109,6 +110,21 @@ func (d *Dev) Duplex() conn.Duplex {
...
 
-//
+// Addr is an I²C slave address.
+type Addr uint16
+
+// Set sets the Addr to a value represented by the string s. Values maybe in
+// decimal or hexadecimal form. Set implements the flag.Value interface.
+func (a *Addr) Set(s string) error {
+	// Allow for only maximum of 10 bits for i2c addresses.
+	u, err := strconv.ParseUint(s, 0, 10)
```
Comment (maruel) [inline conn/i2c/i2c.go]: I hadn't realized you could use 10 bits for ParseUint!

**Pull request merged**